app/form/expense_transaction_form.rb

```diff
@@ -10,4 +12,12 @@ class ExpenseTransactionForm < TransactionForm
     validates :category_id, presence: true
     validates :comment, length: {maximum: 80}
   end
+
+  def amount_valid?
+    errors.add(:amount, I18n.t('insufficient_funds')) if category.amount < amount.to_f
+  end
+
+  def category
```

Comment:
@marderer let's make `#category` method private. It's needed only for this class